lib/index.coffee

```diff
@@ -51,7 +53,38 @@ streamToString = (stream) ->
 		.on 'end', ->
 			resolve(Buffer.concat(chunks).toString())
 
+describeCounterMetricIfNotExists = (metrics, name, description, labelNames) ->
+	if metrics.exists(name) == 'false'
+		metrics.describe.counter(name, description,
+			{
+				labelNames: labelNames
+			}
+		)
+
+describeHistogramMetricIfNotExists = (metrics, name, description, labelNames, buckets) ->
+	if metrics.exists(name) == 'false'
+		metrics.describe.histogram(name, description,
+			{
+				labelNames: labelNames,
+				buckets: buckets
+			}
+		)
+
+setupMetrics = (metrics) ->
```

Comment:
Let's do this simply with `_`

```
setupMetrics = (metrics) ->
  _.once(() ->
    metrics.describe.counter(...)
    metrics.describe.counter(...)
    metrics.describe.counter(...)
    metrics.describe.histogram(...)
    ...
  )
```

Reply:
I tried that, but the problem is that the Builder can create n different GC objects (minimum 3). So if I describe a metric more than once I get an exception. I could either dynamically create the metric name as a combination of gc_<metric description>_<base unit>_<host name> or use a label. From our conversation in FD, we decided to go for the latter :blush:

Reply:
Didn't realized the lodash function! Didn't know about this one, works like a charm. Thanks!